### DISCUSSION TITLE
Is there python way to interact with graphql in an Apollo client way?

### DISCUSSION THREAD

**Discussion by itsklimov** [strawberry-graphql/strawberry#3624] [Q&A] [answered]
In typescript, I use apolloclient to interact with parsed graphql schema and then make calls via it in a very simple way, like on the picture bellow. 

Does python have similar tools?

https://www.apollographql.com/docs/react/development-testing/static-typing/
![image](https://github.com/user-attachments/assets/0e5cdeda-528e-4a95-9991-602be37411f5)

**Comment by DoctorJohn** [comment, marked as answer]
Yes, there are various Python GraphQL clients available. Some of them also support generating type-safe client code. I would recommend going through the following page to find a package you like.

https://graphql.org/community/tools-and-libraries/?tags=python_client
